projects/dsp-ui/src/lib/viewer/views/resource-view/resource-view.component.ts

```diff
@@ -240,6 +251,64 @@ export class ResourceViewComponent implements OnInit, OnChanges, OnDestroy {
         }
     }
 
+    /**
+     * Updates the standoff link value for the resource being displayed.
+     *
+     */
+    private _updateStandoffLinkValue() {
+
+        if (this.resource === undefined) {
+            return;
```

Comment:
That should actually never occur: if the user is able to click on a standoff link, then the resource must have been initialised before in resource-view comp.

I added a comment in https://github.com/dasch-swiss/dsp-ui-lib/pull/228/commits/76e251a4069bbf82d6c4589fdb1d18265986a7fa

However, this is an easy way to know that `this.resource.id` won't cause an error and that no malformed query will be sent to Knora.

If the method is left immediately, then simply the standoff link value would not be updated.

I implemented `DisplayEditComponent._getStandoffLinkValueForResource` so it would not fail if it cannot find the standoff link value, see: https://github.com/dasch-swiss/dsp-ui-lib/blob/81b1e4332cb2b90c462b4ffe452472deaa1f256c/projects/dsp-ui/src/lib/viewer/operations/display-edit/display-edit.component.ts#L157-L191

Now if it cannot find a corresponding standoff link value, there will simply be no event emitted:

https://github.com/dasch-swiss/dsp-ui-lib/blob/81b1e4332cb2b90c462b4ffe452472deaa1f256c/projects/dsp-ui/src/lib/viewer/operations/display-edit/display-edit.component.ts#L193-L224

This could be the case because of timing. The standoff link values are only queried after the values have been updated, and it is also async. And it that case the user would just click again and by then the values should be there.

Reply:
I will write a test that makes sure that `display-edit` does not break when there is no corresponding standoff link value

Reply:
@mdelez I added two tests in https://github.com/dasch-swiss/dsp-ui-lib/pull/228/commits/3812be673e01cf2cb07c64387cdb689e8ea89b36